### ISSUE TITLE
Deleting a stream using gRPC produces an incorrect error

### ISSUE THREAD

**Issue by andriykohut** [EventStore/EventStore#3547]
**Describe the bug**
Calling `Streams.Delete` for a stream that doesn't exist or using an unknown revision number produces incorrect error message:

```
StatusCode.FAILED_PRECONDITION
details = "Append failed due to WrongExpectedVersion. Stream: ..., Expected version: -1, Actual version: 0"
```

**To Reproduce**
1. Use `Streams.Delete` on an unknown stream with `Stream exists` expected state, or call a delete on an existing stream with revision that doesn't exist

**Expected behavior**
The error should mention Delete and not Append

**Config/Logs/Screenshots**
Log example:
```
es-client-eventstore.db-1         | [    1,14,20:29:33.633,INF] Error status code 'FailedPrecondition' raised.
es-client-eventstore.db-1         | Grpc.Core.RpcException: Status(StatusCode="FailedPrecondition", Detail="Append failed due to WrongExpectedVersion. Stream: 3aadc515-4027-4823-89bf-b7e0f92f6a0e, Expected version: -1, Actual version: 0")
es-client-eventstore.db-1         |    at EventStore.Core.Services.Transport.Grpc.Streams`1.DeleteInternal(String streamName, Int64 expectedVersion, ClaimsPrincipal user, Boolean hardDelete, Boolean requiresLeader, CancellationToken cancellationToken) in /home/runner/work/TrainStation/TrainStation/build/oss-eventstore/src/EventStore.Core/Services/Transport/Grpc/Streams.Delete.cs:line 152
es-client-eventstore.db-1         |    at EventStore.Core.Services.Transport.Grpc.Streams`1.Delete(DeleteReq request, ServerCallContext context) in /home/runner/work/TrainStation/TrainStation/build/oss-eventstore/src/EventStore.Core/Services/Transport/Grpc/Streams.Delete.cs:line 48
es-client-eventstore.db-1         |    at Grpc.Shared.Server.UnaryServerMethodInvoker`3.AwaitInvoker(Task`1 invokerTask, GrpcActivatorHandle`1 serviceHandle)
es-client-eventstore.db-1         |    at Grpc.Shared.Server.UnaryServerMethodInvoker`3.AwaitInvoker(Task`1 invokerTask, GrpcActivatorHandle`1 serviceHandle)
es-client-eventstore.db-1         |    at Grpc.AspNetCore.Server.Internal.CallHandlers.UnaryServerCallHandler`3.HandleCallAsyncCore(HttpContext httpContext, HttpContextServerCallContext serverCallContext)
es-client-eventstore.db-1         |    at Grpc.AspNetCore.Server.Internal.CallHandlers.ServerCallHandlerBase`3.<HandleCallAsync>g__AwaitHandleCall|8_0(HttpContextServerCallContext serverCallContext, Method`2 method, Task handleCall)
es-client-eventstore.db-1         | [    1,55,20:29:34.212,DBG] "streamExistenceFilter" is flushing at 562,536. Diff 9,455 ...
es-client-eventstore.db-1         | [    1,55,20:29:34.390,INF] Flushed 131 pages out of 33,334. 1,073,152 bytes. Delay 128 ms per batch. Total delay 128 ms. Actively flushing: 00:00:00.0477977 22.45 MB/s.
es-client-eventstore.db-1         | [    1,55,20:29:34.391,DBG] "streamExistenceFilter" has flushed at 562,536. Diff 9,455. Took 00:00:00.1786115
es-client-eventstore.db-1         | [    1,49,20:29:39.398,DBG] "streamExistenceFilter" took checkpoint at position: 562,536.
```

**EventStore details**
 - EventStore server version: 21.10.6.0

 - Operating system: running an official image in docker

**Comment by ylorph** [comment]
You' right! Thanks for reporting  this .

**Comment by timothycoleman** [comment]
This should be fixed in the upcoming 22.10.0 release. I'll close this for now please feel free to re-open if you have further problems
Thanks!
